torch/utils/_triton.py

```diff
@@ -0,0 +1,17 @@
+import functools
+
+import torch
+
+from torch._inductor.cuda_properties import get_device_capability
+
+
+@functools.lru_cache(None)
+def has_triton() -> bool:
+    if not torch.cuda.is_available():
+        return False
+    try:
+        import triton
```

Comment:
For usage outside of inductor, this form would be more suitable as it avoids eagerly importing `triton` which can be quite slow.
https://github.com/pytorch/pytorch/blob/f092eecc9221ea5048fd11304b3ba5166589d963/torch/cuda/__init__.py#L1262

Reply:
Running `import triton` results in adding `trition` global symbol into `torch.utils._trition` package, which seems OK until somebody starts relying on it...

Reply:
> Running import triton results in adding trition global symbol into torch.utils._trition package, which seems OK until somebody starts relying on it...

I don't think this is true.  `import X` in the scope of a function does not create a global variable, it is function local.  

It does populate `sys.modues["triton"]` though.


```
>>> def foo():
...   import triton
... 
>>> foo()
>>> triton
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'triton' is not defined
>>> 
```